### PR TITLE
🔧 Update .pre-commit-config

### PR DESCRIPTION
**Pull request by trallard**
Split from #792 

## Description

- **:wrench: Update .pre-commit-config**
- **:wrench: Update pre-commit config in pyproject.toml**
- **Update docs and actions to use lint**

## Pull request checklist

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

## How to test

- Run pre-commit locally with `pre-commit run all-files`
- Run with hatch with `hatch env run -e lint lint`

### CODE REVIEW

**Review by peytondmurray — APPROVED** [review]
Looks great - I especially like that pre-commit.ci will autoupdate. :rocket: 

Something weird is up with one of the tests, though it doesn't look like anything related. Is this flaky?

**Comment by trallard** [comment]
Seems like one of those network issues. I tried and re run the workflow multiple times with no avail and  there is nothing in this PR that would affect tests. 

Since it's a parallel test one will tag @nkaretnikov

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -7,34 +7,28 @@
...
-  - repo: https://github.com/pycqa/isort
-    rev: 5.12.0
-    hooks:
-      - id: isort
-        name: isort
-        additional_dependencies: [toml]
-        files: \.py$
-        args: ["--profile", "black"]
-        exclude: "conda-store-server/conda_store_server/action/__init__.py"
-
```
Comment (nkaretnikov) [inline .pre-commit-config.yaml:35]: Why is this removed? Is it assumed that pre-commit will handle this?

I reordered some imports in `conda-store-server/conda_store_server/api.py`, but `hatch env run -e lint lint` didn't fix that.

**Review by nkaretnikov — APPROVED** [review]
@trallard That was a flaky test. Approving to unblock, but see the inline question about import sorting.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (trallard) [reply .pre-commit-config.yaml:35]: I had swapped to the `isort` in `ruff` but it makes little difference to be honest and if it makes it easier/less confusing to keep the sort pre-commit it is ok

**Comment by trallard** [comment]
Thanks @nkaretnikov - those flaky tests are an absolute pain. 

Reverted to vanilla `isort` btw
